pegasus/sites.v3/code.org/public/js/volunteer-local.js

```diff
@@ -0,0 +1,325 @@
+    $('#volunteer-search-facets').show();
+    $('#controls').html('');
+  } else {
+    displayNoResults();
+  }
+  $('#volunteer-search-results').show();
+
+  loadMap(locations);
+}
+
+function getLocations(results) {
+  var locations = [];
+
+  if(results.response){
+    var volunteers = results.response.docs; // The actual volunteers that were returned by Solr.
+    var volunteers_count = volunteers.length;
+
+    for(var i = 0; i < volunteers_count; i++){
+      var index = i;
+      var coordinates = volunteers[i].location_p.split(',');
+      var lat = coordinates[0];
+      var lon = coordinates[1];
+      var title = volunteers[i].name_s;
+      var id = volunteers[i].id;
+      var html = compileHTML(index, volunteers[i]);
+      var contact_title = compileContact(index, volunteers[i]);
+      var contact_link = '<a id="contact-trigger-' + index + '" class="contact-trigger" onclick="return contactVolunteer()">Contact</a>';
+
+      var location = {
+        lat: lat,
+        lon: lon,
+        title: title,
+        id: id,
+        contact_title: contact_title,
+        html: html + contact_link,
+        zoom: 10
+      };
+
+      locations.push(location);
+    }
+  }
+
+  return locations;
+}
+
+function resetFacets() {
+  $.each(selectize, function(key, select) {
+    select.selectize.clear();
+    select.selectize.refreshOptions(false);
+  });
+}
+
+function updateFacets(results) {
+  var facet_fields = results.facet_counts.facet_fields;
+}
+
+function displayNoResults() {
+  $('#controls').html('<p>No results were found.</p>');
+
+  // Hide the facets by default.
+  $('#volunteer-search-facets').hide();
+
+  // If a facet has a value, show the facets.
+  var form_data = $('#volunteer-search-form').serializeArray();
+  $.each(form_data, function(key, field) {
+    if (field.name != 'location' && field.value) {
+      $('#volunteer-search-facets').show();
+    }
+  });
+}
+
+function displayQueryError() {
+  $('#volunteer-search-facets').hide();
+  $('#volunteer-search-results').hide();
+  $('#volunteer-search-error').html('<p>An error occurred. Please try your search again.</p>').show();
+}
+
+function loadMap(locations) {
+  var coordinates = gmap_loc.split(',');
+  var lat = coordinates[0];
+  var lng = coordinates[1];
+
+  // Reset the map.
+  $('#gmap').html('');
+  gmap = new Maplace;
+
+  var map_options = {
+    map_options: {
+      set_center: [lat, lng],
+      zoom: 12
+    },
+    controls_type: 'list',
+    controls_on_map: false
+  };
+
+  if (locations.length > 0) {
+    map_options.force_generate_controls = true;
+    map_options.locations = locations;
+    map_options.afterOpenInfowindow = function(index, location, marker) {
+      setContactTrigger(index, location, marker);
+    };
+  }
+
+  gmap.Load(map_options);
+}
+
+function compileHTML(index, location) {
+  var lines = [];
+  var line;
+
+  // Compile HTML.
+  var html = '<h3>' + location.name_s + '</h3>';
+
+  if (location.company_s) {
```

Comment:
I don't think hound is supposed to be bothering us with js errors so I think you can ignore this